common/lib/driver-definitions/api-report/driver-definitions.api.md

```diff
@@ -139,8 +139,8 @@ export interface IDocumentService {
 
 // @public (undocumented)
 export interface IDocumentServiceFactory {
-    createContainer(createNewSummary: ISummaryTree | undefined, createNewResolvedUrl: IResolvedUrl, logger?: ITelemetryBaseLogger): Promise<IDocumentService>;
-    createDocumentService(resolvedUrl: IResolvedUrl, logger?: ITelemetryBaseLogger): Promise<IDocumentService>;
+    createContainer(createNewSummary: ISummaryTree | undefined, createNewResolvedUrl: IResolvedUrl, logger?: ITelemetryBaseLogger, props?: ILoaderInfoProps): Promise<IDocumentService>;
+    createDocumentService(resolvedUrl: IResolvedUrl, logger?: ITelemetryBaseLogger, props?: ILoaderInfoProps): Promise<IDocumentService>;
```

Comment:
If we pass the logger, then there is no need to pass additional props, as same can be done by creating child logger, right?

Reply:
But I was trying to use the same path to provide info to connectDeltaStream, to pass loaderVersion to IConnect message to convey to push.

Reply:
Ah, got it! Why not put it as optional arg on connectToDeltaStream() API?
There are couple other choices if it's only for ordering service:

IClient.IClientDetails.environment (passing IClient on connectToDeltaStream)
IClient.user.<Some field>

We do not populate environment on calling connectToDeltaStream(), so that seems like great place to represent anything client wants to communicate to ordering service RE it's environment (which I'd think should include loader & driver versions, i.e. each of these layers will append some sub-string before passing it to ordering service).

@GaryWilber, @tanviraumi - what do you think?

runtime version would be great to have too, but there might be no runtime yet when we connect, and there is no convenient way to pass this info, so I'd skip for now.

Reply:
Would we want the all users in a session to know about the telemetry properties the client is sending? If we add it to client details, it would become part of the document state. So it would be sent to other clients and stored in the orderer & summaries.
If we add it to just IConnected, it would only be used for logging in the connect document flow and other clients won't know about it.

Reply:
Yes it will be part of summary. But is that problematic? Putting loaderVersion in clientDetails could make sense as to tell this client was on which loader version? @vladsud @GaryWilber @tanviraumi

Reply:
IConnected is moving info down (from service to client), so that's not useful. We need IConnect, right?
Client can still use IClient to pass this info to driver on connectToDeltaStream() (same way as we use IClient.mode -> IConnect.mode fill). Driver may chose to either clear environment on IClient or not.